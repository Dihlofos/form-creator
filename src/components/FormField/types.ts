import { RadioField } from 'ui/Radio/types';

export type InputTypes = {
  type:
    | 'text'
    | 'number'
    | 'textarea'
    | 'checkbox'
    | 'date'
    | 'radio'
    | 'submit'
    | 'button'
    | 'reset';
};

export type Field = {
  title?: string;
  id: string;
  label?: string;
  name: string;
  placeholder?: string;
  value?: string;
  checked?: boolean;
  defaultValue?: string;
  defaultChecked?: boolean;
  radioValues?: RadioField[];
  min?: number;
  max?: number;
} & InputTypes;

export type FormFieldProps = {
  field: Field;
  value: string | boolean;
  checked?: string | boolean;
  onChange: (e: any) => void;
};
