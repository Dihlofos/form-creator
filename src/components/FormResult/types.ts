export type FormLocalState = Record<string, string | boolean>;
