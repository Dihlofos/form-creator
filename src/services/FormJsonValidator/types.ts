export type FormStructureSchema = {
  title: string;
  action: string;
  method: string;
  items: {
    type:
      | 'text'
      | 'number'
      | 'textarea'
      | 'checkbox'
      | 'date'
      | 'radio'
      | 'submit'
      | 'button'
      | 'reset';
    title: string;
    id: string;
    label: string;
    name: string;
    placeholder: string;
    value: string;
    defaultValue: string;
    defaultChecked: boolean;
    radioValues: { label: string; value: string }[];
    min: number;
    max: number;
  }[];
  buttons: {
    id: string;
    label: string;
    name: string;
    action: string;
  }[];
};
