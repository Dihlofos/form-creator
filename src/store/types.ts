import { Field } from 'components/FormField/types';
import { Button } from 'ui/Button/types';

export type FormStructure = {
  title?: string;
  items: Field[];
  buttons?: Button[];
};

export type FormState = {
  formData: FormStructure;
  activeTab: 'config' | 'result';
};

export type Store = {
  form: FormState;
};
