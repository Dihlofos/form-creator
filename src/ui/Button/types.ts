export type Button = {
  id?: string;
  label?: string;
  name?: string;
  action?: string;
};
