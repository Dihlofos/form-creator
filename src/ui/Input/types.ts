export type OwnProps = {
  label?: string;
  value?: string | boolean;
};
