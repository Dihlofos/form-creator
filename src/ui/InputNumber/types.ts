export type OwnProps = {
  label?: string;
};
