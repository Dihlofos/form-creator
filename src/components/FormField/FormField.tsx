import React, { FC } from 'react';
import { Input, TextArea, Checkbox, Radio } from 'ui';
import { FormFieldProps } from './types';

const FormField: FC<FormFieldProps> = ({ field, onChange, value, checked }) => {
  const {
    id,
    type,
    title,
    label,
    placeholder,
    name,
    defaultChecked,
    defaultValue,
    radioValues,
  } = field;

  if (type === 'text' || type === 'date' || type === 'number') {
    return (
      <Input
        id={id}
        type={type}
        label={label}
        placeholder={placeholder}
        name={name}
        value={String(value)}
        defaultValue={defaultValue}
        onChange={onChange}
      />
    );
  }

  if (type === 'radio' && radioValues) {
    return (
      <>
        {title && <h4>{title}</h4>}
        <Radio
          id={id}
          type={type}
          name={name}
          value={value}
          radioValues={radioValues}
          onChange={onChange}
        />
      </>
    );
  }

  if (type === 'checkbox') {
    return (
      <Checkbox
        id={id}
        type={type}
        name={name}
        checked={Boolean(checked)}
        defaultChecked={defaultChecked}
        onChange={onChange}
      >
        {label}
      </Checkbox>
    );
  }

  if (type === 'textarea') {
    return (
      <TextArea
        id={id}
        placeholder={placeholder}
        label={label}
        name={name}
        value={String(value)}
        defaultValue={defaultValue}
        onChange={onChange}
      />
    );
  }

  return null;
};

export default FormField;
