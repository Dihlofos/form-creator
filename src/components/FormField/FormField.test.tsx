import React from 'react';
import { create } from 'react-test-renderer';
import FormField from './FormField';
import { FormFieldProps } from './types';

describe('ContentBox testing', () => {
  const props: FormFieldProps = {
    field: {
      defaultValue: 'Женя Самсонов',
      id: 'name',
      label: 'Имя',
      name: 'name',
      placeholder: 'Введите имя',
      type: 'text',
    },
    value: 'Женя Самсонов',
    onChange: () => {},
  };

  it('snapshot matches', () => {
    const tree = create(<FormField {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
