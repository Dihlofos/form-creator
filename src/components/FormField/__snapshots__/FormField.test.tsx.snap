// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBox testing snapshot matches 1`] = `
<div
  className="wrapper-input-component"
>
  <div
    className="ant-row ant-form-item label-input-component"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="name"
        title="Имя"
      >
        Имя
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        />
      </div>
    </div>
  </div>
  <input
    className="ant-input input-component"
    id="name"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="Введите имя"
    type="text"
    value="Женя Самсонов"
  />
</div>
`;
